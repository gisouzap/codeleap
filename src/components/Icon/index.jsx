import PropTypes from 'prop-types';

import edit from './icons/edit.svg';
import trash from './icons/delete.svg';

function Icon({ name, ...props }) {
  const icons = { edit, trash };

  return <img src={icons[name]} {...props} />;
}

Icon.propTypes = {
  name: PropTypes.string.isRequired,
};

export default Icon;
