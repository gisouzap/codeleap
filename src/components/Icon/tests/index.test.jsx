import React from 'react';
import { render } from '@testing-library/react';
import Icon from '../index';

describe('Icon', () => {
  it('should render and match the snapshot', () => {
    const spy = jest.spyOn(global.console, 'error');

    const { container } = render(<Icon name="edit" title="Edit" />);

    expect(container).toMatchSnapshot();
    expect(spy).not.toHaveBeenCalled();
  });
});
