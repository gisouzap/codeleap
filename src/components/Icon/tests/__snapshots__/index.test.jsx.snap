// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render and match the snapshot 1`] = `
<div>
  <img
    src="test-file-stub"
    title="Edit"
  />
</div>
`;
